include/operators/binary_arithmetic.hpp

```diff
@@ -156,6 +156,14 @@ BigInt BigInt::operator*(const BigInt& num) const {
     BigInt product;
     if (abs(*this) <= FLOOR_SQRT_LLONG_MAX and abs(num) <= FLOOR_SQRT_LLONG_MAX)
         product = std::stoll(this->value) * std::stoll(num.value);
+    else if (is_power_of_10(this->value)){ // if LHS is a power of 10 do optimised operation 
+        product = num;
```

Comment:
The following will be much faster:
```C++
product.value = num.value;
```
(`product.sign` is taken care of before returning).